### PR TITLE
Migrate bicep example: 101/function-premium-vnet-integration

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/101/function-premium-vnet-integration

The corresponding PR for the modified bicep example is here: https://github.com/Azure/bicep/pull/3171

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
FYI @mcollier you're welcome to review if you'd like.

File: quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+@description('The name of the function app that you wish to create.')
+param appName string = 'fnapp${uniqueString(resourceGroup().id)}'
+
+@allowed([
+  'Standard_LRS'
+  'Standard_GRS'
+  'Standard_RAGRS'
+])
+@description('Storage Account type')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep:23]: move to the top

File: quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
+          value: appInsights.properties.InstrumentationKey
+        }
+        {
+          name: 'APPLICATIONINSIGHTS_CONNECTION_STRING'
+          value: 'InstrumentationKey=${appInsights.properties.InstrumentationKey}'
+        }
+        {
+          name: 'AzureWebJobsStorage'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};EndpointSuffix= ${environment().suffixes.storage};AccountKey=${listkeys(storageAccount.id, '2021-04-01').keys[0].value}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep:116]: I think should we follow the same pattern here (as in the others) and use ```storageAccount.apiVersion``` for the api version

File: quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        }
+        {
+          name: 'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccountName};EndpointSuffix=${environment().suffixes.storage};AccountKey=${listkeys(storageAccount.id, '2021-04-01').keys[0].value};'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep:120]: and here...

File: quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  dependsOn: [
+    virtualNetwork
+  ]
+}
+
+resource networkConfig 'Microsoft.Web/sites/networkConfig@2020-06-01' = {
+  parent: function
+  name: 'virtualNetwork'
+  properties: {
+    subnetResourceId: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, subnetName)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep:146]: subnet.id ?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep (hunk 1 above)
Comment (StephenWeatherford) [reply quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep:23]: decompiler now does this automatyically.

**Pull request merged**